xarray/core/computation.py

```diff
@@ -1137,3 +1137,109 @@ def where(cond, x, y):
                        join='exact',
                        dataset_join='exact',
                        dask='allowed')
+
+
+def _gradient_once(variable, coord, edge_order):
+    """ Compute the gradient along 1 dimension for Variable.
+    variable, coord: Variable
+    """
+    from .variable import Variable
+
+    result_array = duck_array_ops.gradient(
+        variable.data, coord.data, edge_order=edge_order,
+        axis=variable.get_axis_num(coord.dims[0]))
+    return Variable(variable.dims, result_array)
+
+
+def gradient(dataarray, coords, edge_order=1):
+    """ Compute the gradient of the dataarray with the second order accurate
+    central differences.
+
+    Parameters
+    ----------
+    dataarray: xr.DataArray
+        Target array
+    coords: str or sequence of strs
+        The coordinate to be used to compute the gradient.
+    edge_order: 1 or 2. Default 1
+        N-th order accurate differences at the boundaries.
+
+    Returns
+    -------
+    gradient: DataArray or a sequence of DataArrays
+
+    See also
+    --------
+    numpy.gradient: corresponding numpy function
+
+    Examples
+    --------
+
+    >>> da = xr.DataArray(np.arange(12).reshape(4, 3), dims=['x', 'y'],
+    ...                   coords={'x': [0, 0.1, 1.1, 1.2]})
+    >>> da
+    <xarray.DataArray (x: 4, y: 3)>
+    array([[ 0,  1,  2],
+           [ 3,  4,  5],
+           [ 6,  7,  8],
+           [ 9, 10, 11]])
+    Coordinates:
+      * x        (x) float64 0.0 0.1 1.1 1.2
+    Dimensions without coordinates: y
+    >>>
+    >>> xr.gradient(da, 'x')
+    <xarray.DataArray (x: 4, y: 3)>
+    array([[30.      , 30.      , 30.      ],
+           [27.545455, 27.545455, 27.545455],
+           [27.545455, 27.545455, 27.545455],
+           [30.      , 30.      , 30.      ]])
+    Coordinates:
+      * x        (x) float64 0.0 0.1 1.1 1.2
+    Dimensions without coordinates: y
+    >>>
+    >>> xr.gradient(da, ('x', 'y'))
```

Comment:
It would be another option, but in order to store them to a dataset, we will need to define their names very heuristically...

Maybe we can drop this api as this does not speed up the computation and it's easy to do the same thing manually.

Reply:
I agree. I don't think it's particularly useful to compute to compute independent gradients with respect to multiple axes at the same time, e.g., `du/dx`, `du/dy`, etc. If anything, I would be more excited about computing higher order derivatives, e.g., `\frac{d^2 u}{dx dy}`